skyportal/handlers/api/instrument.py

```diff
@@ -72,10 +78,14 @@ def get(self, instrument_id=None):
             _ = Telescope.get_if_owned_by(instrument.telescope_id,
                                           self.current_user)
             return self.success(data=instrument)
+
+        inst_name = self.get_query_argument("name", None)
         query = Instrument.query.filter(Instrument.telescope_id.in_(
             DBSession().query(GroupTelescope.telescope_id).filter(GroupTelescope.group_id.in_(
                 [g.id for g in self.current_user.accessible_groups]
             ))))
+        if inst_name is not None:
+            query = query.filter(Instrument.name == inst_name)
```

Comment:
This returns a list when someone requests an instrument by name, instead of a scalar. Is that intended?

Reply:
@dannygoldstein yes, in order to be consistent with the docs, any GET `/api/instrument` request must return an array of instruments (query parameters do not change this), while `/api/instrument/<instrument_id>` returns a scalar. We went back and forth for a while about whether to implement a new `/api/instrument/byName/<instrument_name>` or similar endpoint that would return a scalar object, but ultimately decided upon this approach.